### PR TITLE
Adjust for renamed app folder, but keep support for ezp 5.x

### PR DESCRIPTION
**Pull request by andrerom**
work in progress, depends on: https://github.com/ezsystems/ezplatform/pull/70

### CODE REVIEW

File: dockerfiles/ezphp/generate_kickstart_file.sh [hunk 1]
```diff
@@ -3,9 +3,9 @@
 # Parameter $1 is value of EZ_KICKSTART_FROM_TEMPLATE ( could be empty )
 
 if [ ! -f ezpublish_legacy/kickstart.ini-dist ]; then
-    if [ -d ezpublish ]; then
-        echo "WARNING: EZ_KICKSTART is set to true, but ezpublish_legacy do not seem to exists"
-        exit 1
+    if [ -d app ]; then
```
Comment (vidarl) [inline dockerfiles/ezphp/generate_kickstart_file.sh]: Kickstart only applies to 5.4 where folder is named ezpublish, right?

File: dockerfiles/ezphp/prepare_distribution_volume.sh [hunk 2]
```diff
@@ -39,8 +39,14 @@ function set_permissions
     if [ ! -d web/var ]; then
         sudo -u ez mkdir web/var
     fi
-    setfacl -R -m u:$APACHE_RUN_USER:rwX -m u:ez:rwX ezpublish/{cache,logs,sessions} web/var
-    setfacl -dR -m u:$APACHE_RUN_USER:rwX -m u:ez:rwX ezpublish/{cache,logs,sessions} web/var
+
+    if [ -d ezpublish ]; then
```
Comment (vidarl) [inline dockerfiles/ezphp/prepare_distribution_volume.sh]: In stead of duplicating the setfacls commands, I would prefere the method used earlier in PR:

```
+APP_FOLDER="app"
+if [ -d ezpublish ]; then
+    APP_FOLDER="ezpublish"
+fi
```

File: dockerfiles/ezphp/run.sh [hunk 3]
```diff
@@ -38,25 +38,37 @@ echo "Setting permissions on eZ Publish folder as they might be broken if rsync
 if [ ! -d web/var ]; then
     sudo -u ez mkdir web/var
 fi
-setfacl -R -m u:$APACHE_RUN_USER:rwX -m u:ez:rwX ezpublish/{cache,logs,sessions} web/var
-setfacl -dR -m u:$APACHE_RUN_USER:rwX -m u:ez:rwX ezpublish/{cache,logs,sessions} web/var
+
+if [ -d ezpublish ]; then
```
Comment (vidarl) [inline dockerfiles/ezphp/run.sh]: In stead of duplicating the setfacls commands, I would prefere the method used earlier in PR:

```
+APP_FOLDER="app"
+if [ -d ezpublish ]; then
+    APP_FOLDER="ezpublish"
+fi
```

File: dockerfiles/internal/nginx/run.sh [hunk 4]
```diff
@@ -11,8 +11,8 @@ cp -a ${BASEDIR}/doc/nginx/etc/nginx/ez_params.d /etc/nginx/
 sed -i "s@  fastcgi_pass unix:/var/run/php5-fpm.sock;@  # fastcgi_pass unix:/var/run/php5-fpm.sock;@" /etc/nginx/conf.d/ez.conf
 sed -i "s@  #fastcgi_pass 127.0.0.1:9000;@  fastcgi_pass php_fpm:${PHP_FPM_PORT_9000_TCP_PORT};@" /etc/nginx/conf.d/ez.conf
 
-# Setting environment for ezpublish ( dev/prod/behat etc )
-sed -i "s@  #fastcgi_param ENVIRONMENT dev;@  fastcgi_param ENVIRONMENT ${EZ_ENVIRONMENT};@" /etc/nginx/conf.d/ez.conf
+# Setting environment for ezplatform ( dev/prod/behat etc )
+sed -i "s@  #fastcgi_param SYMFONY_ENV dev;@  fastcgi_param SYMFONY_ENV ${EZ_ENVIRONMENT};@" /etc/nginx/conf.d/ez.conf
```
Comment (vidarl) [inline dockerfiles/internal/nginx/run.sh]: Ain't working on 5.4

File: Vagrantfile [hunk 5]
```diff
@@ -115,13 +115,13 @@ Vagrant.configure("2") do |config|
       ./build.sh
       ./docker-compose_ezpinstall.sh
       ./docker-compose.sh up -d --no-recreate
-      if [[ ( -d volumes/ezpublish/ezpublish ) && ( ! -d volumes/ezpublish/ezpublish_legacy ) ]];  then \
+      if [[ ( -f volumes/ezpublish/app/config/ezplatform.yml ) ]];  then \
```
Comment (vidarl) [inline Vagrantfile]: Do we still have to support 5.4 in Vagrantfile?

File: dockerfiles/ezphp/config/xdebug.ini [hunk 6]
```diff
@@ -11,5 +11,5 @@ xdebug.remote_port=9001
 
 xdebug.profiler_enable = 1
 xdebug.profiler_output_name = xdebug.out.%t
-xdebug.profiler_output_dir = /var/www/ezpublish/log
+xdebug.profiler_output_dir = /var/www/app/log
```
Comment (vidarl) [inline dockerfiles/ezphp/config/xdebug.ini]: Do we need to support 5.4 in this xdebug.ini ?

File: dockerfiles/ezphp/generate_kickstart_file.sh (hunk 1 above)
Comment (andrerom) [inline dockerfiles/ezphp/generate_kickstart_file.sh]: yes, hence why I did not do a error anymore when app folder is there.

File: dockerfiles/ezphp/prepare_distribution_volume.sh (hunk 2 above)
Comment (andrerom) [inline dockerfiles/ezphp/prepare_distribution_volume.sh]: feel free :)

File: dockerfiles/ezphp/run.sh (hunk 3 above)
Comment (andrerom) [inline dockerfiles/ezphp/run.sh]: yep

File: dockerfiles/internal/nginx/run.sh (hunk 4 above)
Comment (andrerom) [inline dockerfiles/internal/nginx/run.sh]: true, so we'll need some logic here.

File: Vagrantfile (hunk 5 above)
Comment (andrerom) [inline Vagrantfile]: doesn't hurt does it? this is now only triggered for platform, while 5.4 and a standard symfony install this is skipped.

File: dockerfiles/ezphp/config/xdebug.ini (hunk 6 above)
Comment (andrerom) [inline dockerfiles/ezphp/config/xdebug.ini]: could, but not a must I guess.

File: dockerfiles/ezphp/prepare_distribution_volume.sh (hunk 2 above)
Comment (vidarl) [inline dockerfiles/ezphp/prepare_distribution_volume.sh]: hmmm.. I didn't notice at first that you also removed "sessions" in the setfacls... So, leaving as-is

File: dockerfiles/ezphp/config/xdebug.ini (hunk 6 above)
Comment (vidarl) [inline dockerfiles/ezphp/config/xdebug.ini]: 5.4 remains supported by adding ```xdebug.ini-ezp54``` file which is used instead when needed

**Comment by vidarl** [comment]
@andrerom  : Ready for review

**Comment by andrerom** [comment]
+1

File: create_distro_containers.sh [hunk 7]
```diff
@@ -274,6 +283,9 @@ function pushonly
 echo parse_commandline_arguments
 parse_commandline_arguments "$@"
 
+echo getAppFolder:
+getAppFolder
```
Comment (andrerom) [inline create_distro_containers.sh]: debug code here? :)

File: resources/ezpublish.yml_varnishpurge.diff [hunk 8]
```diff
@@ -1,5 +1,5 @@
---- ezpublish/config/ezpublish.yml.org	2015-01-20 14:32:40.287422725 +0000
-+++ ezpublish/config/ezpublish.yml	2015-01-20 14:32:56.191423023 +0000
+--- app/config/ezplatform.yml.org	2015-01-20 14:32:40.287422725 +0000
++++ app/config/ezplatform.yml	2015-01-20 14:32:56.191423023 +0000
```
Comment (andrerom) [inline resources/ezpublish.yml_varnishpurge.diff]: be aware the patch probably does not apply anymore

File: create_distro_containers.sh (hunk 7 above)
Comment (vidarl) [inline create_distro_containers.sh]: not really, I log to stdout before every function call..

This is to easier see where things go wrong if they do in prod though..

File: resources/ezpublish.yml_varnishpurge.diff (hunk 8 above)
Comment (vidarl) [inline resources/ezpublish.yml_varnishpurge.diff]: Yeah. I just consider the varnish changes as best-effort. The varnish containers must anyway be reworked before we start using them

**Pull request merged**